### ISSUE TITLE
Pandas Dataframe.explode function parameter TypeError 

### ISSUE THREAD

**Issue by RomanSutter** [microsoft/pylance-release#3871]
## Environment data

-   Language Server version: 2023.1.40
-   OS and version: Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): Poetry, Python 3.9.12

## Code Snippet

Pandas Version: 1.5.2
```python
data = {'col1': [1,2,3,4], 'col2': [5,6,7,8]}
df = pd.DataFrame(data=data)

# Test 1 shows error
exploded_rawdata = df.explode(list(df.columns))

# Test 2 shows error
exploded_rawdata = df.explode(['col1', 'col2'])

```

## Repro Steps
Use pylance while implementing the above code snippet. Pylance marks the parameter "column" as TypeError .

## Expected behavior

No errors about the list used to specify the columns to explode.
This feature is used based on the [pandas documentation](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.explode.html) (parameter column, non-empty list of strings in this case).

## Actual behavior
**Executing the code snippet works as intented, however the following error from pylance is shown:**

Argument of type "list[IndexIterScalar | tuple[Hashable, ...]]" cannot be assigned to parameter "column" of type "_str | tuple[Unknown, ...]" in function "explode"
  Type "list[IndexIterScalar | tuple[Hashable, ...]]" cannot be assigned to type "_str | tuple[Unknown, ...]"
    "list[IndexIterScalar | tuple[Hashable, ...]]" is incompatible with "_str"
    "list[IndexIterScalar | tuple[Hashable, ...]]" is incompatible with "tuple[Unknown, ...]"

## Logs
![Image of the error shown](https://user-images.githubusercontent.com/45934817/214801741-849b41e0-8d8a-4379-89be-2ce5f74d40ad.png)

## Clarification
Is this problem known, or are there any steps to prevent this error (besides the obvious "#type ignore", which only supresses the issue )?

**Comment by rchiodo** [comment]
Thanks for the issue. 

This is happening because the panda type stub has this for explode:

```python
    def explode(self, column: _str | tuple, ignore_index: _bool = ...) -> DataFrame: ...
```

This means it expects a `str` or a `tuple`, not a list. 

My guess is that there's an overload that takes a list and that overload is not declared in the type stub. 

We get panda type stubs from here:
https://github.com/pandas-dev/pandas-stubs

You could log an issue there or add the list type to the explode stub here:
https://github.com/pandas-dev/pandas-stubs/blob/caf9134c7af18a87f64d819a9954be77404ab6f6/pandas-stubs/core/frame.pyi#L1014

**Comment by rchiodo** [comment]
I added a PR to pandas-stubs to try and fix it:
https://github.com/pandas-dev/pandas-stubs/pull/523

**Comment by bschnurr** [comment]
This issue has been fixed in prerelease version 2023.2.31, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023231-15-february-2023-prerelease)
